File: platforms/ios/runtime/Sources/NativeblocksRuntime/experiment/ExperimentUseCase.swift
import Foundation

internal final class ExperimentUseCase {

    private let runtimeClient: NativeRuntimeClientManager

    init(runtimeClient: NativeRuntimeClientManager) {
        self.runtimeClient = runtimeClient
    }

    func get(key: String, cacheTTL: Int64?) async -> (value: String, variableType: String)? {
        guard let result = try? await runtimeClient.experimentClient.getExperiment(key: key, cacheTtl: cacheTTL) else {
            return nil
        }
        return (result.value, result.variableType)
    }
}

File: platforms/android/runtime/core/src/main/java/io/nativeblocks/runtime/experiment/ExperimentUseCase.kt
package io.nativeblocks.runtime.experiment

import io.nativeblocks.runtime.engine.NativeEngineClientManager

internal class ExperimentUseCase(
    private val engineClient: NativeEngineClientManager,
) {

    suspend fun get(key: String, cacheTTL: Long): Pair<String, String>? =
        runCatching {
            val result = engineClient.experimentClient.getExperiment(key, cacheTTL)
            result.value to result.variableType
        }.getOrNull()
}
